ci-operator/step-registry/upi/deprovision/aws/stage1/upi-deprovision-aws-stage1-commands.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+set +e
+
+function queue() {
+    local TARGET="${1}"
+    shift
+    local LIVE
+    LIVE="$(jobs | wc -l)"
+    while [[ "${LIVE}" -ge 45 ]]; do
+    sleep 1
+    LIVE="$(jobs | wc -l)"
+    done
+    echo "${@}"
+    if [[ -n "${FILTER:-}" ]]; then
+    "${@}" | "${FILTER}" >"${TARGET}" &
+    else
+    "${@}" >"${TARGET}" &
+    fi
+}
+
+
+export PATH=$PATH:/tmp/shared
+AWS_SHARED_CREDENTIALS_FILE=${CLUSTER_PROFILE_DIR}/.awscred
+export AWS_SHARED_CREDENTIALS_FILE
+
+echo "Deprovisioning cluster ..."
+export PATH="${HOME}/.local/bin:${PATH}"
+
+AWS_REGION=$(cat ${SHARED_DIR}/AWS_REGION)
+AWS_DEFAULT_REGION=${AWS_REGION}  # CLI prefers the former
+export AWS_REGION
+export AWS_DEFAULT_REGION
+CLUSTER_NAME=$(cat ${SHARED_DIR}/CLUSTER_NAME)
+
+aws --region $AWS_REGION cloudformation describe-stack-resources --stack-name "${CLUSTER_NAME}-control-plane" \
+    --query 'StackResources[?ResourceType==`AWS::EC2::Instance`].PhysicalResourceId' --output text | sed 's,\t,\n,g' > /tmp/node-provider-IDs
+for INDEX in 0 1 2
+do
+    aws --region $AWS_REGION cloudformation describe-stack-resources --stack-name "${CLUSTER_NAME}-compute-${INDEX}" \
+    --query 'StackResources[].PhysicalResourceId' --output text | cut -d, -f1 >> /tmp/node-provider-IDs
+done
+
+while IFS= read -r i; do
+    mkdir -p "${ARTIFACT_DIR}/nodes/${i}"
+    queue ${ARTIFACT_DIR}/nodes/$i/console aws ec2 get-console-output --instance-id "${i}"
```

Comment:
The `documentation` for this step is talking about deprovisioning, and this seems orthogonal.  There's also an existing `gather-aws-console` step that seems like it will do what you're looking for here, so perhaps you can drop this portion and plug `ref: gather-aws-console` into your chains instead?